crates/optimism/primitives/src/supervisor.rs

```diff
@@ -0,0 +1,44 @@
+//! This is our custom implementation of validator struct
+
+use jsonrpsee::http_client::{HttpClient, HttpClientBuilder};
+use std::time::Duration;
+use url::Url;
+
+pub use kona_interop::{ExecutingDescriptor, SafetyLevel};
+pub use kona_rpc::{InteropTxValidator, InteropTxValidatorError};
+
+/// Implementation of the supervisor trait for the interop.
+#[derive(Debug, Clone)]
+pub struct SupervisorClient {
+    client: HttpClient,
+    safety: SafetyLevel,
+}
```

Comment:
same as mentioned in rbuilder pr, we want this to be a generic client, so we can use WsClient for example if we like

Reply:
We want it to be like <T>?
This is a concrete implementation I use, if we want to have something generic maybe let's move it into Kona?

Reply:
then we can just use `alloy::Provider` (e.g. DynProvider) which type erases the transport and still supports sending rpc requests

Reply:
But it won't implement the SupervisorApiClient for the trait, no?

Reply:
all jsonrpsee client traits are blanket implemented for all jsonrpsee client types

Reply:
Okay, I'll try

Reply:
I have made changes, now using SupervisorClient from kona
Now it using ReqwestCient which is not generic tho